### PR TITLE
Add jit request API and schema.

### PR DESCRIPTION
**Pull request by dandanwang0320**
### Description:
Add schema of JIT request related APIs.
Most APIs are from AzureUX-ARM/src/providers/Roles/Providers.Feature/Controllers/JitRequestController.cs 
And "updateAccess" api from
ARM/src/providers/Roles/Providers.Feature/Controllers/ApplicationApiController.cs

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/6185 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/6185/commits/a5e64370baf2986d86db33db077398b7312b62a6

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]
The new swagger isn't referenced in any README.md file so it won't get included in any SDK.  Is this expected?

**Comment by dandanwang0320** [comment]
> The new swagger isn't referenced in any README.md file so it won't get included in any SDK. Is this expected?

**Comment by jhendrixMSFT** [comment]
Suppressions are added only if there's a linter error that can't be fixed for some legitimate reason, so no need to add anything there at present.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json [hunk 1]
```diff
@@ -0,0 +1,1889 @@
...
+      "readOnly": true,
+      "enum": [
+        "NotSpecified",
+        "Pending",
+        "Approved",
+        "Denied",
+        "Failed",
+        "Canceled",
+        "Deleted",
+        "Canceled",
```
Comment (jhendrixMSFT) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json]: Duplicate entry needs to be removed.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json [hunk 2]
```diff
@@ -0,0 +1,1889 @@
...
+            }
+          ],
+          "responses": {
+            "200": {
+              "description": "OK - Returns an array of JIT requests.",
+              "schema": {
+                "$ref": "#/definitions/JitRequestDefinitionListResult"
+              }
+            },         
+            "404": {
```
Comment (jhendrixMSFT) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json]: This indicates that a 404 should be treated as a success, is that intentional?  ARM feedback will be interesting here.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json [hunk 3]
```diff
@@ -0,0 +1,1889 @@
...
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        } 
+      }     
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Solutions/jitRequests": {
+      "get": {
```
Comment (jhendrixMSFT) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json]: Extra "get" should be removed.

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json (hunk 2 above)
Comment (dandanwang0320) [reply specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json]: I was following the convention of previous versions and other ARM APIs. Is there a general guideline for 404?

**Comment by jhendrixMSFT** [comment]
There are model validation failures that need to be fixed, please see the log [here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=28578).

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json (hunk 2 above)
Comment (jhendrixMSFT) [reply specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json]: I've not seen this before so it was a surprise to me, doesn't necessarily mean it's incorrect.  The ARM reviewer can provide a definitive answer.

**Comment by dandanwang0320** [comment]
> There are model validation failures that need to be fixed, please see the log [here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=28578).

Actually none of those warnings were add my new API or schema. Or is there anything needs to be added for existing apis in new version?

**Comment by jhendrixMSFT** [comment]
They will need to be cleaned up at some point, if you don't want to clean them up now (e.g. tight deadline) we can skip it however can you please open an issue to track fixing them?

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json (hunk 2 above)
Comment (dandanwang0320) [reply specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json]: Do you know who's the ARM reviewer? If it's from ARM team, I could talk to them offline.

Comment (jhendrixMSFT) [reply specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json]: Looks like one hasn't been assigned yet.  If this is urgent you can email the ARM on-call person (there are instructions on how to find this info in the ARM API Review Checklist comment at the top.

**Comment by sanjaiganesh** [comment]
Requested for description .. updating with old and new apis contents to see the diff easily.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2018-06-01/managedapplications.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
   "swagger": "2.0",
   "info": {
     "title": "ApplicationClient",
-    "version": "2018-06-01",
+    "version": "2019-03-01",
```
[the remark this answers is not in the document]
Comment (dandanwang0320) [reply specification/resources/resource-manager/Microsoft.Solutions/stable/2018-06-01/managedapplications.json]: Changing it back.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json [hunk 5]
```diff
@@ -0,0 +1,1885 @@
+{
```
Comment (KrisBash) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json]: In the future, please copy the last reviewed API version into the new version folder as the first commit. This enables an expedited review as we can diff the changes between commits.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json [hunk 6]
```diff
@@ -0,0 +1,1885 @@
...
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true
```
Comment (KrisBash) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json]: examples?

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json [hunk 7]
```diff
@@ -0,0 +1,1885 @@
...
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Solutions/applications/{applicationName}/updateAccess": {
+      "post": {
+        "tags": [
```
Comment (KrisBash) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json]: Is there not a request body?

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json [hunk 8]
```diff
@@ -0,0 +1,1885 @@
...
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - Returns the JIT request",
+            "schema": {
+              "$ref": "#/definitions/JitRequestDefinition"
+            }
+          },
+          "404": {
```
Comment (KrisBash) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json]: The `default` error is sufficient, there is no need to model specific error codes

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json [hunk 9]
```diff
@@ -0,0 +1,1885 @@
...
+          "description": "The provider's role definition identifier. This role will define all the permissions that the provider must have on the managed application's container resource group. This role definition cannot have permission to delete the resource group."
+        }
+      },
+      "required": [
+        "principalId",
+        "roleDefinitionId"
+      ],
+      "description": "The managed application provider authorization."
+    },
+    "ErrorResponse": {
```
Comment (KrisBash) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json]: This does not appear to align to RPC error contract, see: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md#error-response-content

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json [hunk 10]
```diff
@@ -0,0 +1,1885 @@
...
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "put": {
```
Comment (KrisBash) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json]: This is modeled as a tracked resource. Thus, it requires a PATCH api that supports update of "tags", but would ideally support update of all mutable properties

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json [hunk 11]
```diff
@@ -0,0 +1,1885 @@
...
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },          
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "201": {
```
Comment (KrisBash) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json]: missing "x-ms-long-running-operation" ?

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json [hunk 12]
```diff
@@ -0,0 +1,1885 @@
...
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "jitRequests"
+        ],
+        "operationId": "jitRequests_CreateOrUpdate",
+        "description": "Creates or updates the JIT request.",
+        "parameters": [
```
Comment (KrisBash) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json]: Where's the request body?

**Review by KrisBash — CHANGES_REQUESTED** [review]
Please see comments for requested changes

**Comment by jhendrixMSFT** [comment]
@dandanwang0320 hello!  Any update here?  Also there's a merge conflict that needs to be resolved.

**Comment by ravbhatnagar** [comment]
@dandanwang0320 - please let me know when this is ready for re-review.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json (hunk 10 above)
Comment (dandanwang0320) [reply specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json]: We don't have a patch api for this resource in our code base. Do you mean we should add that to our back end code?

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json (hunk 9 above)
Comment (dandanwang0320) [reply specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json]: This actually comes with original schema, not changed by this PR.

**Comment by dandanwang0320** [comment]
@KrisBash Hi, would you have time to take another look?

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json (hunk 10 above)
Comment (KrisBash) [reply specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json]: Yes - portal/cli will not be able to update tags on the resource without it

**Comment by NullMDR** [comment]
@openapibot sdkautomation rebuild

**Comment by jhendrixMSFT** [comment]
@dandanwang0320 hello is this ready for review by ARM?

**Comment by yungezz** [comment]
@dandanwang0320 can we close the PR for now since you're not ready for review, and the PR is over several months. Pls create new PR when you're ready for review and merge.

**Comment by jhendrixMSFT** [comment]
Closing due to inactivity.

**Pull request closed without merging**